budget_control_expense/models/account_move_line.py

```diff
@@ -9,7 +9,7 @@ class AccountMoveLine(models.Model):
     def uncommit_expense_budget(self):
         """For vendor bill in valid state, do uncommit for related expense."""
         Expense = self.env["hr.expense"]
-        for ml in self:
+        for ml in self.filtered(lambda l: not l.move_id.not_affect_budget):
```

Comment:
What if, system is set as following,

* Journal Vendor Bill = Not Affect Budget

And then, doing a normal Expense, will also not uncommit budget too. (But in fact, we want to uncommit for case expense).

Reply:
I made the fix into the branch we are working one. https://github.com/OCA/account-budgeting/pull/51/commits/bfdcbdf45e11648df7f88f8cce1875dfc5dbfb00

Let me know what you think. If you test it OK, you can close this. Thanks!